src/UI/Actions.hs

```diff
@@ -1515,13 +1513,11 @@ pipeCommand' cmd
         runExceptT (mkConfig ent >>= runEntityCommand)
           >>= either assignError (const $ pure ())
 
-editAttachment :: AppState -> IO AppState
-editAttachment s =
-    case L.listSelectedElement $ view (asCompose . cAttachments) s of
-        Nothing -> pure $ setError (GenericError "No file selected to edit") s
-        Just (_, m) -> case preview (headers . contentDisposition . folded . dispositionType) m of
-          (Just Inline) -> invokeEditor' s
-          _ -> pure $ setError (GenericError "Not implemented. See #182") s
+editAttachment :: (MonadState AppState m, MonadIO m, MonadMask m) => m ()
+editAttachment = selectedItemHelper (asCompose . cAttachments) $ \m -> do
```

Comment:
Full disclosure: I really hate the `selectedItemHelper` function ;) I've come to hate it while hacking on #5 since the refactoring is fiddly ...

Reply:
I refactored `selectedItemHelper`!  What do you think of the new and improved version?  I find it reasonably pleasant and intuitive now that it is `MonadState` based.

Reply:
I like it! Specifically the part that it doesn't use `AppState -> AppState` as a parameter any more :)